### DISCUSSION TITLE
PDM python version requirement

### DISCUSSION THREAD

**Discussion by trkwyk** [Homebrew/discussions#3989] [Tap maintenance and Homebrew development] [answered]
Hello I'm not sure is this the right place to put this problem though. The package manager PDM for python requires python>=3.7 as per stated in the [`pyproject.toml`](https://github.com/pdm-project/pdm/blob/378a170f3bafcd474da505c3b768888e25c34898/pyproject.toml#L8) whereas the current formulae requires [`python@3.11`](https://formulae.brew.sh/formula/pdm#default).

I'd like not to install too many pythons on my machine so if it's easy to fix would you please give it a fix? Or it does require `python@3.11` and I'm missing something? In that case please let me know.

Thank you for any help you can offer.

**Comment by SMillerDev** [comment, marked as answer]
Homebrew software will always require the latest possible version of their dependencies. the python 3.11 dependency is correct.

**Reply by trkwyk** [reply]
I see. Well in that case if I want to make python 3.11 the default `python` and `python3` (also pip), should I use `brew link python@3.11`? And what should I do to let brew take care of the symlinks? Will `brew cleanup` remove them automatically if one day I `brew uninstall python@3.11`? (Note brew installs just `python3.11` and no `python3` or `python` in `/usr/local/bin`)
